model/src/main/java/org/cbioportal/model/util/Select.java

```diff
@@ -0,0 +1,75 @@
+package org.cbioportal.model.util;
```

Comment:
I think what I am struggling with here is that the Select class should be used in presistence, service and web modules. Is there a more appropriate location that is a common dependency for these modules?

Reply:
A while back, I tried to capture our module dependencies here : https://docs.google.com/document/d/1r3fJYmtzTj5qCC_xI9SkgI6k8N5TQuHnobmx75WAHa0/edit
Note .. out of date .. there is no more business module for instance.

Any class needed for the persistence layer must reside either in the persistence-api module or in the model module. If this is an argument type accepted by the persistence layer api, then it is correct to place it in the model module with the other classes which define the types accepted/delivered by the persistence-api. If we refactored our code base to be more clean, I think I would not share model classes up and down the persistence / service / web layers, but that is a separate discussion. My concern about this class is that it is not a plain/pojo data structure, but it has attached semantics. That is not the case with the bulk of the model module classes. Classes which have semantic functionality probably should be unit tested .. and I see that there is a SelectTest.java class in the core package. We don't need unit tests for the pojo classes in model, but for this class a test makes sense. I just would locate the test class in the same module as the tested class - so I request the test class be relocated to the model module.

Also, if this is an argument type accepted by the persistence layer api then I don't consider it a util (which I think of as helper classes which are only used within the module). So I would relocate it into package org.cbioportal.model and out of org.cbioportal.model.util

Reply:
I do not know what to do here.

You proposed solution of moving the SelectTest does not work because unit tests placed in the _model_ module do not run. That is the reason for me putting the SelectTest in the core module.

When following your explanation I understand that the model module is for plain POJO models for the persistence layer. In that case I agree that Select class should not be placed there. But what to do with classes that are shared by basically all modules in the project? The Select class basically is used like java.util.List. Would this justify a new module that is used by all other modules?

I am open to any alternative solution.

Reply:
Let's reach out to someone else to give input on this question. I wonder if @kalletlak or @Luke-Sikina have an opinion on the question of where it would be best to locate a class such as Select. (which module, and which package) This is a kind of wrapper class with some utility functions and semantics --- it is a type which is accepted as an argument by functions in the persistence-api interface. It has unit tests, but those are currently (in this PR) located in the core module because the model module currently is not configured to run unit tests. Any ideas on properly locating this class (and its tests)? I've expressed a mixed opinion in this PR but I'm uncertain.

Reply:
Since this a utility used across different module, may be it should be in its own module. which also makes testing easy. I would prefer to have it in a new module(something like utility/utils).

Reply:
Thank you for the comment @kalletlak. If this were in a separate module, it would need to be added as one of the dependencies for the persistence-api module (in pom.xml)
Any thoughts on whether it is "ok" to be creating a new module to hold classes which have semantics which need testing and also are used in the persistence layer API interface? This is the first such class ... perhaps there will be others to come?

Reply:
Also note : this relocation may be done after the merge of this PR (as a post-merge refactoring task)

Reply:
@sheridancbio I think it should be ok to create new module. Also we could move any existing utilities to this new module